projects/rero/ng-core/src/lib/record/editor/extensions.ts

```diff
@@ -70,13 +70,33 @@ export class NgCoreFormlyExtension {
    * @param field - FormlyFieldConfig
    */
   onPopulate(field: FormlyFieldConfig) {
+    // Tranfer default value on templateOptions and remove on field
+    if (field.type === 'string' && field.defaultValue) {
+      field.templateOptions.defaultValue = field.defaultValue;
+      delete field.defaultValue;
+    }
+
     this._hideShowEmptyField(field);
     this._setWrappers(field);
     field.options.fieldChanges.subscribe(changes => {
-      if (changes.type === 'expressionChanges' && changes.property === 'templateOptions.required' && changes.value === true) {
-        if (changes.field.hide === true) {
-          changes.field.hide = false;
-        }
+      // Set default value on field if i have a default value into the templateOptions
+      if (
+        changes.type === 'hidden'
+        && field.templateOptions
+        && field.templateOptions.defaultValue
+        && !field.hide
+        && (!field.model || !Object.keys(field.model).includes(String(field.key)))
+      ) {
+        field.formControl.setValue(field.templateOptions.defaultValue);
+      }
```

Comment:
It could be interesting to explain all conditions used. If I well understand, it fill the default value from schema if the field was hidden (now visible) and if related model doesn't already have this value inside ?
I think It could be very of interest to include maximum documentation on how it works for next changes/developpers.